### PR TITLE
Documenting CPS changes in Visual Studio 2019 Preview 1

### PR DESCRIPTION
**Pull request by adrianvmsft**

### CODE REVIEW

File: doc/overview/breaking_changes_visual_studio_2019.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+The following interfaces no longer inherit from `IProjectVersionedValue`:
+
+| Interface |
+| --- |
+| IProjectSnapshot |
+| IProjectRuleSnapshot |
+| IProjectSharedFoldersSnapshot |
+| IProjectImportTreeSnapshot |
+
+The project version is already carried in the dataflow. This is a breaking change.
```
[the remark this answers is not in the document]
Comment (lifengl) [reply doc/overview/breaking_changes_visual_studio_2019.md]: Also, the reason behind this change is to have a consistent style in all dataflow services, and also reuse unchanged immutable objects in some cases intead of forcing to create new instances to include version only changes.

File: doc/Index.md [hunk 2]
```diff
@@ -85,4 +85,4 @@ VS Project System Documentation
   - [Add Single File Generator](scenario/add_single_file_generator.md)
 - [Differences between Visual Studio 2013 and Visual Studio 2015](overview/VS2013_vs_VS2015.md)
 - [Changes](../CHANGES.md)
-- [Breaking Changes in Visual Studio 2017](overview/breaking_changes_visual_studio_next.md)
+- [Breaking Changes in Visual Studio 2017](overview/breaking_changes_visual_studio_2017.md)
```
Comment (davilimap) [inline doc/Index.md]: We should also add a line for breaking changes in VS 2019

File: doc/Index.md [hunk 3]
```diff
@@ -85,4 +85,4 @@ VS Project System Documentation
   - [Add Single File Generator](scenario/add_single_file_generator.md)
 - [Differences between Visual Studio 2013 and Visual Studio 2015](overview/VS2013_vs_VS2015.md)
 - [Changes](../CHANGES.md)
```
Comment (davilimap) [inline doc/Index.md]: While were here, we should move this item up and maybe rename it for clarity

File: doc/automation/obtaining_the_MSBuild.Project_from_CPS.md [hunk 4]
```diff
@@ -7,6 +7,23 @@ Obtaining the MSBuild.Project from CPS
...
    use the MSBuild Project object exclusively within the lock:
 
+Visual Studio 2019
+```csharp
+await projectLockService.WriteLockAsync(
+    async access =>
+    {
+        Project project = await access.GetProjectAsync(configuredProject);
+
+        // party on it, respecting the type of lock you've acquired.
```
Comment (davilimap) [inline doc/automation/obtaining_the_MSBuild.Project_from_CPS.md]: Revise the beginning of this sentence?

File: doc/Index.md [hunk 5]
```diff
@@ -85,4 +85,4 @@ VS Project System Documentation
   - [Add Single File Generator](scenario/add_single_file_generator.md)
 - [Differences between Visual Studio 2013 and Visual Studio 2015](overview/VS2013_vs_VS2015.md)
```
Comment (davilimap) [inline doc/Index.md]: Nit: rename this to _Breaking Changes in Visual Studio 2015 as below_

File: doc/Index.md (hunk 2 above)
Comment (adrianvmsft) [reply doc/Index.md]: I created a parent node and grouped them together

File: doc/Index.md (hunk 3 above)
Comment (adrianvmsft) [reply doc/Index.md]: I grouped all breaking changes together

File: doc/overview/breaking_changes_visual_studio_2019.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    },
+    cancellationToken);
+```
+
+## New Slim DataFlow blocks
+
+To reduce the memory usage of dataflow blocks, CPS introduces a new custom set of slim dataflow blocks corresponding to `TransformBlock`/`ActionBlock`/`BroadcastBlocks`.
+While this is not a breaking change, we recommend CPS extensions to start using the new blocks.
+
+The new blocks allow CPS dataflow to skip middle versions easier, but may introduce some behavior changes.
```
Comment (drewnoakes) [inline doc/overview/breaking_changes_visual_studio_2019.md]: In what cases would a version be skipped? My mental model is that all updates are queued. Is there some conflation going on? Is this only for the slim versions, or did the prior blocks do this too?

**Pull request merged**